### PR TITLE
Rework `add_grub_bootloader()`

### PR DESCRIPTION
**Pull request by codefiles**
- This fixes issue: #1493 (#881)

## PR Description:

- Rework `add_grub_bootloader()`
- Replace `sed` approach with Pythonic style
- Disable zswap if zram is enabled

## Tests and Checks
- [ ] I have tested the code!

### CODE REVIEW

File: archinstall/lib/installer.py [hunk 1]
```diff
@@ -888,19 +888,48 @@ def add_grub_bootloader(self, boot_partition :Partition, root_partition :Partiti
...
+		options = [f'rootfstype={root_fs_type}']
+
+		if self._zram_enabled:
+			options.insert(0, 'zswap.enabled=0')
+
 		if real_device := self.detect_encryption(root_partition):
 			root_uuid = SysCommand(f"blkid -s UUID -o value {real_device.path}").decode().rstrip()
-			_file = "/etc/default/grub"
-			add_to_CMDLINE_LINUX = f"sed -i 's/GRUB_CMDLINE_LINUX=\"\"/GRUB_CMDLINE_LINUX=\"cryptdevice=UUID={root_uuid}:cryptlvm rootfstype={root_fs_type}\"/'"
-			enable_CRYPTODISK = "sed -i 's/#GRUB_ENABLE_CRYPTODISK=y/GRUB_ENABLE_CRYPTODISK=y/'"
```
Comment (Torxed) [inline archinstall/lib/installer.py:895]: I haven't used GRUB in a millennia. But I do agree that explicitly creating a config file is probably better.
But are there other options in the default conf that we would lose if we don't use the `sed` approach?

Way cleaner code the way this PR suggests, just checking in if we're losing anything by doing it?

Comment (codefiles) [reply archinstall/lib/installer.py:895]: I also do not use GRUB. I did a minimal amount of testing on the code using the `grub.default` file, which is distrusted as `/etc/default/grub`, from the package source for GRUB on Arch Linux.

This should produces an identical copy of the `/etc/default/grub` file as the `sed` approach does and is just the pythonic way of doing it.

**Comment by Dungeonseeker** [comment]
Couldn't you simply echo the parameter into "/etc/sysctld.d/99-zram.conf"? That way it would work regardless of what bootloader the user is running.

**Comment by codefiles** [comment]
That is certainly a valid option and one I had not considered. I was simply replicating the solution that was merged for systemd-boot.

If the pull requests that I have open for boot loaders are merged I plan to then submit a pull request to move any of the code that is duplicated in the boot loader functions to the `add_bootloader()` function. This would include moving kernel parameters to `add_bootloader()` and adding them as a parameter to the boot loader functions.

**Review by Torxed — APPROVED** [review]

**Comment by Torxed** [comment]
Looks good and the future plans are a nice direction. Just needs the latest `rework-grub` fix and conflicts sorted and I'll merge in!

**Comment by codefiles** [comment]
I plan to close this once the kernel parameters are taking care of in a separate pull request since most of this was superseded by:

- #1962
- #1968

**Comment by codefiles** [comment]
In addition to the list given in the last comment, also superseded by:
- #2079

**Pull request closed without merging**